### PR TITLE
Ko makes an effect

### PR DESCRIPTION
**Pull request by KellyOgilvie**
Casino! The flicker issue fix I noted in my previous commit has been fixed in jdb-bgcolor.

### CODE REVIEW

File: python/mp/effects/__init__.py [hunk 1]
```diff
@@ -4,5 +4,6 @@
 from mp.effects import throb
 from mp.effects import bounce
 from mp.effects import set_color
+from mp.effects import casino
 
-__all__ = ['effect', 'set_color', 'sine_wave', 'throb', 'bounce']
+__all__ = ['effect', 'set_color', 'sine_wave', 'throb', 'bounce', 'casino']
```
Comment (jdblair) [inline python/mp/effects/__init__.py:9]: I would like to see if we can get rid of this list. Its one less thing to maintain when we add effects.

File: python/mp/effects/casino.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        
+
+    def next(self, rosary):
+        super().next()
+
+        #turn on all beads from end_position to end of bead set
+        for b in range(self.end_position, len(self.bead_list)):
+            self.bead_list[b].color.set(self.color)
+        
+        self.bead_list[self.last].color.set(rosary.bgcolor)
```
Comment (jdblair) [inline python/mp/effects/casino.py]: You don't need to set the "last" bead to bgcolor anymore. Now we set all the beads to bgcolor at the top of the mainloop, so you know is already cleared. I would expect this to cause flicker.

File: TODO.org [hunk 3]
```diff
@@ -72,3 +72,4 @@
 *** TODO mount boards
 *** TODO mount connectors
 *** TODO label and close it up
+** TODO design and assemble the equipment cabinet
```
Comment (jdblair) [inline TODO.org:75]: whoops, i think this was pulled in when you merged my jdb-bgcolor branch. sorry about that.

File: python/mp/effects/casino.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    dm = copy.deepcopy(effect.Effect.dm)
+
+    def __init__(self, bead_set, color=color.Color(), duration=None, direction=1):
+        super().__init__("casino", bead_set, color=color, duration=duration)
+        self.direction = direction
+        if (self.direction < 0):
+            self.current = len(self.bead_list) - 1
+        else:
+            self.current = 0
+        self.last = self.current
```
Comment (jdblair) [inline python/mp/effects/casino.py]: you should be able to get rid of self.last now. see comment below.

File: python/server.py [hunk 5]
```diff
@@ -55,7 +55,7 @@ def trigger_something(unused_addr):
     # Since the Rosary itself won't be instantiated often, I don't feel
     # bad about requiring that the dispatcher be passed
     r = rosary.Rosary(args.ip, args.port, d)
-    r.start(interactive=False)
+    r.start(interactive=True)
```
Comment (jdblair) [inline python/server.py:58]: crap, this was also from my jdb-bgcolor

**Review by jdblair — COMMENTED** [review]
Please look into removing the use of self.last. You shouldn't need this anymore, and the code will be easier to understand without it.

File: python/mp/effects/__init__.py (hunk 1 above)
Comment (jdblair) [reply python/mp/effects/__init__.py:9]: Nevermind, experimentation suggests we do need it to work this way.

**Comment by KellyOgilvie** [comment]
I'm closing this in favour of the new one i'm about to submit. I really hope that's the right thing to do in this situation. YOLO

**Pull request closed without merging**